tests/functional/object/mcg/ui/test_bucket_policy_ui.py

```diff
@@ -0,0 +1,232 @@
+
+            # Wait for new OBC bucket to appear (up to 30 seconds)
+            target_bucket_name = None
+            for attempt in range(6):  # 6 attempts * 5 seconds = 30 seconds max wait
+                current_buckets = set(bucket_ui.get_buckets_list())
+                new_buckets = current_buckets - initial_buckets
+
+                # Look for newly created bucket with OBC pattern
+                obc_buckets = [
+                    b for b in new_buckets if b.startswith("test-bucket-obc-")
+                ]
+
+                if obc_buckets:
+                    target_bucket_name = obc_buckets[0]  # Use first OBC bucket found
+                    break
+
+                if attempt < 5:  # Don't sleep on last attempt
+                    time.sleep(5)
+
+            if not target_bucket_name:
+                # Fallback: try to find any bucket with OBC pattern in the current list
+                all_buckets = bucket_ui.get_buckets_list()
+                obc_buckets = [
+                    b for b in all_buckets if b.startswith("test-bucket-obc-")
+                ]
+
+                if obc_buckets:
+                    target_bucket_name = obc_buckets[0]
+                else:
+                    pytest.skip("No OBC bucket found for account-dependent policy test")
+        else:
+            # For policies that don't require account ID, use any existing bucket
+            buckets = bucket_ui.get_buckets_list()
+            if not buckets:
+                pytest.skip("No buckets available for testing")
+
+            target_bucket_name = buckets[0]
+
+        bucket_permissions_ui = BucketsTabPermissions()
+
+        policy_method = getattr(bucket_permissions_ui, method_name)
+        policy_method(bucket_name=target_bucket_name, **params)
+
+        logger.info(f"Successfully completed {policy_name} bucket policy test")
+
+    @polarion_id("OCS-6893")
+    def test_delete_bucket_policy_ui(self, setup_ui_class_factory):
+        """
+        Test deleting bucket policy via UI.
+
+        This test uses the unified delete workflow that:
+        1. Navigates to Object Storage and bucket permissions
+        2. Verifies a policy exists before attempting deletion
+        3. Activates policy editor and deletes the policy
+        4. Handles confirmation dialog automatically
+
+        Note: This test assumes a policy already exists from previous tests.
+
+        Raises:
+            pytest.skip: If no buckets are available for testing
+        """
+        setup_ui_class_factory()
+        logger.info("Starting test to delete bucket policy")
+
+        bucket_ui = BucketsTab()
+        bucket_ui.navigate_buckets_page()
+
+        buckets = bucket_ui.get_buckets_list()
+        if not buckets:
+            pytest.skip("No buckets available for testing")
+
+        bucket_permissions_ui = BucketsTabPermissions()
+
+        # Use unified delete workflow
+        bucket_permissions_ui.delete_bucket_policy_ui(bucket_name=None)
```

Comment:
Won't this always use first.bucket? I may have missed a part of the setup, but it shouldn't have a bucket policy to begin with...

IMO it'll be safer to create a bucket, set a policy on it and then delete it. Other tests might rely on it not having a policy, or temper with it in some other way before this test.